sdcm/utils/perftune_validator.py

```diff
@@ -0,0 +1,197 @@
+import json
+import logging
+import random
+import traceback
+import yaml
+
+from sdcm.sct_events import Severity
+from sdcm.sct_events.system import PerftuneResultEvent
+
+
+# https://docs.scylladb.com/stable/operating-scylla/admin-tools/perftune.html
+
+
+PERFTUNE_LOCATION = "/opt/scylladb/scripts/perftune.py"
+TEMP_PERFTUNE_YAML_PATH = "/tmp/perftune.yaml"
+PERFTUNE_EXPECTED_RESULTS_PATH = "defaults/perftune_results.json"
+
+
+def get_machine_architecture_type(node):
+    result = node.remoter.run("uname -m")
+    return result.stdout.strip()
+
+
+def get_number_of_cpu_cores(node) -> int:
+    result = node.remoter.run("grep -c ^processor /proc/cpuinfo")
+    cores_num = int(result.stdout)
+    return cores_num
+
+
+def get_default_mode(node):
+    number_of_cores = get_number_of_cpu_cores(node=node)
+    if number_of_cores <= 4:
+        return "mq"
+    elif number_of_cores <= 8:
+        return "sq"
+    elif number_of_cores <= 32:
+        return "sq_split"
+    return None
```

Comment:
@ShlomiBalalis this part is wrong.
What's need to be done here is to read the file and check which mode was set.

We can add a validation with similar logic as above to verify that the correct mode is set.